packages/client/lib/miner/miner.ts

```diff
@@ -0,0 +1,254 @@
+import { BlockHeader } from '@ethereumjs/block'
+import { BN } from 'ethereumjs-util'
+import { ConsensusType } from '@ethereumjs/common'
+import { Event } from '../types'
+import { Config } from '../config'
+import { FullSynchronizer } from '../sync'
+
+export interface MinerOptions {
+  /* Config */
+  config: Config
+
+  /* FullSynchronizer */
+  synchronizer: FullSynchronizer
+}
+
+/**
+ * @module miner
+ */
+
+/**
+ * Implements Ethereum block creation and mining.
+ * @memberof module:miner
+ */
+export class Miner {
+  private DEFAULT_PERIOD = 15
+  private config: Config
+  private synchronizer: FullSynchronizer
+  private assembling: boolean
+  private period: number
+  public running: boolean
+
+  /* global NodeJS */
+  private _nextAssemblyTimeoutId: NodeJS.Timeout | undefined
+
+  /**
+   * Create miner
+   * @param options constructor parameters
+   */
+  constructor(options: MinerOptions) {
+    this.config = options.config
+    this.synchronizer = options.synchronizer
+    this.running = false
+    this.assembling = false
+    this.period = (this.config.chainCommon.consensusConfig().period ?? this.DEFAULT_PERIOD) * 1000 // defined in ms for setTimeout use
+  }
+
+  /**
+   * Convenience alias to return the latest block in the blockchain
+   */
+  private latestBlockHeader(): BlockHeader {
+    return (this.synchronizer as any).chain.headers.latest
```

Comment:
Would this ever not be the `fullSynchronizer`?  Just wondering why it's casted as `any`?

Reply:
`chain` in `FullSynchronizer` is declared as private so just trying to get access to it :/